plenum/test/helper.py

```diff
@@ -534,15 +534,20 @@ def stopNodes(nodes: List[TestNode], looper=None, ensurePortsFreedUp=True):
         assert looper, 'Need a looper to make sure ports are freed up'
     for node in nodes:
         node.stop()
+
+    ports = [[n.nodestack.ha[1], n.clientstack.ha[1]] for n in nodes]
```

Comment:
Why do this computation before you know that "port freeness" needs to be ensured? And why burden `waitUntillPortIsAvailable` with the concern that it has to flatten the list of ports, lets keep it simple and do the flattening here